charms/argo-server/tox.ini

```diff
@@ -31,11 +31,11 @@ allowlist_externals =
     find
     pip-compile
     xargs
-commands = 
-; uses 'bash -c' because piping didn't work in regular tox commands
-  pip-compile requirements.in
-  pip-compile requirements-fmt.in
-  bash -c 'find . -type f -name "requirements*.in" | xargs --replace=\{\} pip-compile --resolver=backtracking \{\}'
+commands =
+    ; we must preserve the order of compilation, since each *.in file depends on some *.txt file.
+    ; For example, requirements-unit.in depends on requirements.txt and we must compile first
+    ; requirements.txt to ensure that requirements-unit.txt get the same dependency as the requirements.txt
+    bash -c 'for pattern in "requirements.in" "requirements-fmt.in" "requirements*.in"; do find . -type f -name "$pattern" -exec bash -c "cd \$(dirname "{}") && pip-compile --resolver=backtracking \$(basename "{}")" \;; done'
```

Comment:
do we need the `cd \$(dirname "{}") &&`?  I think tox is always running from the root dir that has the requirements files right?  Seemed to work if I removed this locally

Reply:
That is working because you executed directly those tox files. However, in other repos you will have tox not in a charm dir but in the repo root.

Then you start recursively calling the command from repo root, which adds paths to the file

And since this is a demo project before cannon run, I would like to agree on the common command and message comment